### PR TITLE
Homework 7 task 3 

### PR DESCRIPTION
**Pull request by ivanzuev78**
Done

### CODE REVIEW

File: HomeWork_7/hw3.py [hunk 1]
```diff
@@ -1,26 +1,49 @@
...
+            return "x"
+        elif all(map(lambda x: x == "o", row)):
+            return "o"
+        return False
+
+    all_lines_to_check_win = (
+        board
+        + [[board[i][j] for i in range(3)] for j in range(3)]
+        + [[board[i][i] for i in range(3)]]
+        + [[board[i][2 - i] for i in range(3)]]
```
Comment (Cjkjvfnby) [inline HomeWork_7/hw3.py:39]: IMHO it is much easier to just write all 8 combination by hand.    
You can leave it as is.

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -29,4 +29,4 @@ repos:
           - flake8-pytest-style==1.3.0
           - flake8-return==1.1.2
         args:
-          - "--ignore=S101"
\ No newline at end of file
+          - "--ignore=S101,W503"
```
[the remark this answers is not in the document]
Comment (ivanzuev78) [reply .pre-commit-config.yaml:32]: I can't understand, what does it mean. Now I know, thanks.

File: HomeWork_7/hw3.py [hunk 3]
```diff
@@ -1,26 +1,49 @@
...
+        [[-, -, o],
+         [-, o, o],
+         [x, x, x]]
+
+         Return value should be "x wins!"
+
+    """
+
+    def check_win(row: List):
+        if all(map(lambda x: x == "x", row)):
```
[the remark this answers is not in the document]
Comment (ivanzuev78) [reply HomeWork_7/hw3.py:29]: Yes, it is much easier.

File: HomeWork_7/hw3.py [hunk 4]
```diff
@@ -1,26 +1,49 @@
...
+            return "x"
+        elif all(map(lambda x: x == "o", row)):
+            return "o"
+        return False
```
[the remark this answers is not in the document]
Comment (ivanzuev78) [reply HomeWork_7/hw3.py]: It was `Union[str, bool]`
It is `str` now.

File: HomeWork_7/hw3.py (hunk 1 above)
Comment (ivanzuev78) [reply HomeWork_7/hw3.py:39]: Agree with you.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**